deepcell_toolbox/utils.py

```diff
@@ -450,3 +450,265 @@ def untile_image(tiles, tiles_info, power=2, **kwargs):
+    tiles_info['dtype'] = image.dtype
+    tiles_info['pad_x'] = pad_x
+    tiles_info['pad_y'] = pad_y
+    tiles_info['pad_z'] = pad_z
+
+    return tiles, tiles_info
+
+
+def window_3D(window_size, overlap_z=(5, 5), overlap_x=(32, 32), overlap_y=(32, 32), power=3):
+    """
+    Make a 1D window function, then infer and return a 2D window function.
+    Done with an augmentation, and self multiplication with its transpose.
+    Could be generalized to more dimensions.
+    """
+    window_z = spline_window(window_size[0], overlap_z[0], overlap_z[1], power=power)
+    window_x = spline_window(window_size[1], overlap_x[0], overlap_x[1], power=power)
+    window_y = spline_window(window_size[2], overlap_y[0], overlap_y[1], power=power)
+
+    window_z = np.expand_dims(np.expand_dims(np.expand_dims(window_z, -1), -1), -1)
+    window_x = np.expand_dims(np.expand_dims(np.expand_dims(window_x, -1), -1), -1)
+    window_y = np.expand_dims(np.expand_dims(np.expand_dims(window_y, -1), -1), -1)
+
+    window = window_z * window_x.transpose(1, 0, 2, 3) * window_y.transpose(1, 2, 0, 3)
+
+    return window
+
+
+def untile_image_3D(tiles, tiles_info, power=3, force=False, **kwargs):
+    """Untile a set of tiled images back to the original model shape.
+
+     Args:
+         tiles (numpy.array): The tiled images image to untile.
+         tiles_info (dict): Details of how the image was tiled (from tile_image).
+         power (int): The power of the window function
+
+     Returns:
+         numpy.array: The untiled image.
+     """
+    # Define mininally acceptable tile_size and stride_ratios for spline interpolation
+    min_tile_size = 32
+    min_stride_ratio = 0.5
+
+    if force:
+        min_tile_size = 0
+        min_tile_height = 0
+        min_stride_ratio = 0
+
+    stride_ratio = tiles_info['stride_ratio']
+    image_shape = tiles_info['image_shape']
+    batches = tiles_info['batches']
+
+    x_starts = tiles_info['x_starts']
+    x_ends = tiles_info['x_ends']
+    y_starts = tiles_info['y_starts']
+    y_ends = tiles_info['y_ends']
+    z_starts = tiles_info['z_starts']
+    z_ends = tiles_info['z_ends']
+
+    overlaps_x = tiles_info['overlaps_x']
+    overlaps_y = tiles_info['overlaps_y']
+    overlaps_z = tiles_info['overlaps_z']
+
+    tile_size_x = tiles_info['tile_size_x']
+    tile_size_y = tiles_info['tile_size_y']
+    tile_size_z = tiles_info['tile_size_z']
+    pad_x = tiles_info['pad_x']
+    pad_y = tiles_info['pad_y']
+    pad_z = tiles_info['pad_z']
+
+    image_shape = [image_shape[0], image_shape[1], image_shape[2], image_shape[3], tiles.shape[-1]]
+    window_size = (tile_size_z, tile_size_x, tile_size_y)
+    image = np.zeros(image_shape, dtype=np.float)
+
+    for (tile, batch, x_start, x_end, y_start, y_end, z_start,
+         z_end, overlap_x, overlap_y, overlap_z) in zip(
+            tiles, batches, x_starts, x_ends, y_starts, y_ends,
+            z_starts, z_ends, overlaps_x, overlaps_y, overlaps_z):
```

Comment:
Save the output of `zip()` as another variable. This is a little tough to read.

Reply:
done